lib/error/error_handler.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Error
+  module ErrorHandler
+    include SentryLogging
+
+    def self.included(klass)
+      klass.class_eval do
+        rescue_from 'Exception' do |e|
+          handle_the_error(e)
+        end
+      end
+    end
```

Comment:
This could be a hidden "gotcha" for those who aren't aware that `Error::ErrorHandler` rescues all exceptions. I'd prefer to see this written out in `ApplicationController` - something like this:

```ruby
# in application_controller.rb
rescue_from 'Exception' do |exception|
  log_the_error(exception)
  va_exception = coerce_exception(exception)
  
  headers['WWW-Authenticate'] = 'Token realm="Application"' if va_exception.is_a?(Common::Exceptions::Unauthorized)
  
  render json: { errors: va_exception.errors }, status: va_exception.status_code
end
```